packages/api/src/command/medical/patient/convert-fhir-bundle.ts

```diff
@@ -0,0 +1,203 @@
+    dateFrom,
+    dateTo,
+    conversionType,
+  });
+
+  return {
+    resourceType: "Bundle",
+    total: 1,
+    type: "collection",
+    entry: [
+      {
+        resource: {
+          resourceType: "DocumentReference",
+          subject: {
+            reference: `Patient/${patient.id}`,
+          },
+          content: [
+            {
+              attachment: {
+                contentType: `application/${conversionType}`,
+                url: url,
+              },
+            },
+          ],
+        },
+      },
+    ],
+  };
+}
+
+async function convertFHIRBundleToMedicalRecord({
+  bundle,
+  patient,
+  resources,
+  dateFrom,
+  dateTo,
+  conversionType,
+}: {
+  bundle: Bundle<Resource>;
+  patient: Pick<Patient, "id" | "cxId" | "data">;
+  resources?: ResourceTypeForConsolidation[];
+  dateFrom?: string;
+  dateTo?: string;
+  conversionType: ConsolidationConversionType;
+}): Promise<string> {
+  const lambdaName = Config.getFHIRToMedicalRecordLambdaName();
+
+  // create file name
+  const fileName = createFileName({
+    cxId: patient.cxId,
+    patientId: patient.id,
+    resources,
+    dateFrom,
+    dateTo,
+  });
+
+  // store on S3
+  const s3 = makeS3Client(Config.getAWSRegion());
+  const emptyMetaProp = "na";
+  await s3
+    .putObject({
+      Bucket: Config.getMedicalDocumentsBucketName(),
+      Key: fileName,
+      Body: JSON.stringify(bundle),
+      ContentType: "application/json",
+      Metadata: {
+        patientId: patient.id,
+        cxId: patient.cxId,
+        resources: resources?.toString() ?? emptyMetaProp,
+        dateFrom: dateFrom ?? emptyMetaProp,
+        dateTo: dateTo ?? emptyMetaProp,
+        conversionType,
+      },
+    })
+    .promise();
+
+  const payload: FhirToMedicalRecordPayload = {
+    fileName,
+    patientId: patient.id,
+    firstName: patient.data.firstName,
+    cxId: patient.cxId,
+    dateFrom,
+    dateTo,
+    conversionType,
+  };
+
+  const lambdaResult = await lambdaClient
+    .invoke({
+      FunctionName: lambdaName,
+      InvocationType: "RequestResponse",
+      Payload: JSON.stringify(payload),
+    })
+    .promise();
+
+  if (lambdaResult.StatusCode !== 200) {
+    throw new MetriportError("Lambda invocation failed", undefined, {
+      lambdaName,
+      status: lambdaResult.StatusCode,
+      log: logResultToString(lambdaResult.LogResult),
+      payload: lambdaResult.Payload?.toString(),
+    });
+  }
+  if (!lambdaResult.Payload) {
+    throw new MetriportError("Payload is undefined", undefined, {
+      lambdaName,
+      status: lambdaResult.StatusCode,
+      log: logResultToString(lambdaResult.LogResult),
+    });
+  }
+
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  // TODO Gotta check for error here
+  const url = lambdaResult.Payload.toString();
```

Comment:
This is the part outstanding, will require update of 2 lambdas and some shared code to be used here.